src/libstore/store-api.cc

```diff
@@ -1104,9 +1104,6 @@ std::map<StorePath, StorePath> copyPaths(
         return storePathForDst;
     };
 
-    // total is accessed by each copy, which are each handled in separate threads
-    std::atomic<uint64_t> total = 0;
```

Comment:
So this didn't have to be global, and is replaced by a per-path `total` in the `sinkToSource` callback

Reply:
It mustn't be global because a local activity is pushed, which tracks the local `total` (done) and `narSize` (expected) sizes already. Thus this was responsible for the insane sizes in the progress bar.